sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.messaging.eventhubs.checkpointstore.jedis;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JsonSerializer;
+import com.azure.core.util.serializer.JsonSerializerProviders;
+import com.azure.core.util.serializer.TypeReference;
+import com.azure.messaging.eventhubs.CheckpointStore;
+import com.azure.messaging.eventhubs.models.Checkpoint;
+import com.azure.messaging.eventhubs.models.PartitionOwnership;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.Jedis;
+
+/**
+ * Implementation of {@link CheckpointStore} that uses Azure Redis Cache, specifically Jedis.
+ */
+public class JedisRedisCheckpointStore implements CheckpointStore {
+    private static final ClientLogger LOGGER = new ClientLogger(JedisRedisCheckpointStore.class);
+    private static final JsonSerializer DEFAULT_SERIALIZER = JsonSerializerProviders.createInstance(true);
+    static final String CHECKPOINT = "checkpoint";
+    static final String  PARTITION_OWNERSHIP = "partitionOwnership";
+    private final JedisPool jedisPool;
+    JedisRedisCheckpointStore(JedisPool jedisPool) {
+        this.jedisPool = jedisPool;
+    }
+
+    /**
+     * This method returns the list of partitions that were owned successfully.
+     *
+     * @param requestedPartitionOwnerships List of partition ownerships from the current instance
+     * @return Null
+     */
+    @Override
+    public Flux<PartitionOwnership> claimOwnership(List<PartitionOwnership> requestedPartitionOwnerships) {
+        return null;
+    }
+
+    /**
+     * This method returns the list of checkpoints from the underlying data store, and if no checkpoints are available, then it returns empty results.
+     *
+     * @param fullyQualifiedNamespace The fully qualified namespace of the current instance of Event Hub
+     * @param eventHubName The Event Hub name from which checkpoint information is acquired
+     * @param consumerGroup The consumer group name associated with the checkpoint
+     * @return Null
+     */
+    @Override
+    public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        try (Jedis jedis = jedisPool.getResource()) {
+            ArrayList<Checkpoint> listStoredCheckpoints = new ArrayList<>();
+            Set<String> members = jedis.smembers(prefix);
+            if (members.isEmpty()) {
+                jedisPool.returnResource(jedis);
```

Comment:
Do we need to explicitly returnResource? I thought that's why the try() {} block existed.

If we do need to returnResource, I don't see it being returned in all other paths..

Reply:
I believe the idea is to return it before the try block ends, since unless you explicitly return the resource the pool assumes that the jedis instance is still in use.

Reply:
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html
The syntactic sugar for try is doing a `close` after the try block ends. . . is this the same as `returnResource`?